### ISSUE TITLE
search bar cuts off the first letter of the query on Android

### ISSUE THREAD

**Issue by jadeynryan** [quarto-dev/quarto-cli#8832]
### Bug description

I'm submitting this issue on behalf of this [comment on LinkedIn](https://www.linkedin.com/feed/update/urn:li:activity:7165733167033778176?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A7165733167033778176%2C7166126002954321921%29&dashCommentUrn=urn%3Ali%3Afsd_comment%3A%287166126002954321921%2Curn%3Ali%3Aactivity%3A7165733167033778176%29). 

A user reported that typing a second letter into the search bar deletes the first letter on an Android mobile device. This occurs on my [website](https://jadeyryan.com/) and others.

Note this only happens with Android mobile:
![android](https://github.com/quarto-dev/quarto-cli/assets/95007373/da084146-73f0-4747-8dc0-6aac725cbf93)

It looks fine on an iPhone:
![iphone](https://github.com/quarto-dev/quarto-cli/assets/95007373/30a943ca-3932-4784-bebb-974a6c44fa3f)

### Steps to reproduce

This occurs on my website and I haven't modified any of the default Quarto search settings. 
Website: <https://jadeyryan.com/>
Repo: <https://github.com/jadeynryan/jadey_website>

### Expected behavior

The entire search query should be shown in the search bar.

### Actual behavior

The search query is cut off.

### Your environment

- IDE: RStudio 2023.12.1 
- OS: Mac Sonoma 14.3.1

### Quarto check output

Quarto 1.4.550
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.550
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.05
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/jadeyryan/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.7 (Conda)
      Path: /opt/anaconda3/bin/python
      Jupyter: 4.8.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.3
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
      knitr: 1.42
      rmarkdown: 2.21

[✓] Checking Knitr engine render......OK

**Comment by cscheid** [comment]
Thanks for the report. We use a third-party library for search, and we have to go report that upstream.

**Comment by dragonstyle** [comment]
I just checked this on a local simulator and I'm unable to reproduce it (FWIW). It likely means that it is related to something specific about the Android version, browser, or behavior. If you have any further details about how you were able to reproduce it, this would be super helpful!

**Comment by seewe** [comment]
This is my android version on a Samsung S23 FE phone.
And this issue happen in chrome browser.


![Screenshot_20240222_104457_Settings](https://github.com/quarto-dev/quarto-cli/assets/40412177/007ea2fa-6672-4cf7-89f7-d2f04106026d)

**Comment by gordonwoodhull** [comment]
I can repro in chrome browser on Samsung Android tablet (but not Firefox on same tablet).

**Comment by dragonstyle** [comment]
I think this is going to be a tough one as I still can't reproduce it on a local simulator trying to emulate the above. @gordonwoodhull If you can repro this, it would be interesting for you to observe the behavior really exactly (e.g. does the `n` in netlify appear then disappear, what happens if you type key by key slowly, etc...) I don't have a great theory just yet about what the cause could be (and there wasn't much in https://github.com/algolia/autocomplete which we use for the typeahead implementation itself)...

**Comment by seewe** [comment]
Yes the apear then desapear. This behavior is not only on the first Word. It's appear sometime at the Middle of the Word. This video shows the behavior. Hope this will help.

https://github.com/quarto-dev/quarto-cli/assets/40412177/9088b2cb-49bb-4c86-9df0-9d5a1a58f16f

**Comment by gordonwoodhull** [comment]
Yes, it's repeatable. The second key always replaces the first key, even after clearing and trying again. 


https://github.com/quarto-dev/quarto-cli/assets/1366709/0d341d03-8806-4094-9bfa-2ffc5e56ea20



Here is a video of me trying to type "netlify" three times. (I don't know where the "p" came from at the start - probably a typo.)

**Comment by dragonstyle** [comment]
Wild! Ok, it really doesn't seem to be a timing issue. More random fishing - does it matter what letter is inserted first (e.g. what if you search for a different term)?

**Comment by gordonwoodhull** [comment]
In addition to the first letter deletion, it seems to be feeding back from the incremental search results, and sometimes deleting the last word if it doesn't match. 

Here I type "things i care about", which does match and gets through after dropping the first letter. 

But then I back up and start typing "becausebecausebecause" and it keeps deleting the last word because the whole phrase does not match. 

https://github.com/quarto-dev/quarto-cli/assets/1366709/e0a57e50-4363-4771-918b-d4b88db97f88

It's more complicated than this, but this is enough to show that the behavior is affected by whether it is matching or not.
